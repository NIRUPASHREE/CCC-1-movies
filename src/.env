REACT_APP_API_KEY=8fa9a5d7460a3c602126d6abdf94f0a9
