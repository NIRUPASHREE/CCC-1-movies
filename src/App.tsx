import { HomePage } from './components/HomePage';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Favorites } from './components/Favorites';

export const App: React.FC = () => {
  return (
    <div className="container mx-auto h-full min-h-full min-w-full bg-black p-8 ">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/home" element={<HomePage />} />
          <Route path="/favorites" element={<Favorites />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
};
