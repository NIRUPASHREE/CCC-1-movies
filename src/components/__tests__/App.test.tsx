import { App } from 'src/App';
import { render, screen } from 'src/utils/test-utils';
import { it, expect } from 'vitest';

describe('Simple working test', () => {
  it('the title is visible', () => {
    render(<App />);
    expect(screen.getByText(/Movies/i)).toBeInTheDocument();
  });
  
  it('renders correctly', () => {
    const result = render(<App />);
    expect(result).toMatchSnapshot();
  });
});
