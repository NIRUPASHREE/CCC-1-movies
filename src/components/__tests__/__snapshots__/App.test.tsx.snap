// Vitest Snapshot v1

exports[`Simple working test > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto h-full min-h-screen min-w-full bg-black p-8"
      >
        <div
          data-testid="home-page"
        >
          <div
            class="flex flex-row justify-between"
            data-testid="header"
          >
            <div
              class="flex flex-row"
            >
              <div
                class="ml-2"
              >
                <svg
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                      fill="#FFF"
                    />
                    <path
                      d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                      fill="#555AB9"
                    />
                    <path
                      d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                      fill="#91BAF8"
                    />
                  </g>
                </svg>
                <h1
                  class="text-white"
                  data-testid="logo-title"
                >
                  Movies
                </h1>
              </div>
              <a
                class="header-buttons is-active"
                data-testid="link-home"
                href="/home"
              >
                HOME
              </a>
              <a
                class="header-buttons"
                data-testid="link-favorites"
                href="/favorites"
              >
                FAVOURITES
              </a>
            </div>
            <div
              class="flex"
            >
              <div
                class="relative mx-auto w-max place-self-center "
              >
                <input
                  class="peer relative z-10 h-12 w-12 cursor-pointer rounded-full border border-transparent bg-transparent pl-12 text-white outline-none duration-300 focus:w-full focus:cursor-text focus:border-white focus:pl-16 focus:pr-4"
                  data-testid="search-bar"
                  name="search"
                  placeholder="movie name.."
                  type="text"
                  value=""
                />
                <svg
                  class="search-icon"
                  fill="none"
                  stroke="white"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="mt-4 w-full"
          >
            <div>
              <div
                class="z-10 flex h-screen items-center justify-center bg-black"
                title="loading"
              >
                <svg
                  class="h-12 w-12 animate-spin"
                  data-testid="loading-icon"
                  fill="none"
                  stroke="white"
                  stroke-width="1"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0l3.181 3.183a8.25 8.25 0 0013.803-3.7M4.031 9.865a8.25 8.25 0 0113.803-3.7l3.181 3.182m0-4.991v4.99"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto h-full min-h-screen min-w-full bg-black p-8"
    >
      <div
        data-testid="home-page"
      >
        <div
          class="flex flex-row justify-between"
          data-testid="header"
        >
          <div
            class="flex flex-row"
          >
            <div
              class="ml-2"
            >
              <svg
                height="32"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z"
                    fill="#FFF"
                  />
                  <path
                    d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z"
                    fill="#555AB9"
                  />
                  <path
                    d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z"
                    fill="#91BAF8"
                  />
                </g>
              </svg>
              <h1
                class="text-white"
                data-testid="logo-title"
              >
                Movies
              </h1>
            </div>
            <a
              class="header-buttons is-active"
              data-testid="link-home"
              href="/home"
            >
              HOME
            </a>
            <a
              class="header-buttons"
              data-testid="link-favorites"
              href="/favorites"
            >
              FAVOURITES
            </a>
          </div>
          <div
            class="flex"
          >
            <div
              class="relative mx-auto w-max place-self-center "
            >
              <input
                class="peer relative z-10 h-12 w-12 cursor-pointer rounded-full border border-transparent bg-transparent pl-12 text-white outline-none duration-300 focus:w-full focus:cursor-text focus:border-white focus:pl-16 focus:pr-4"
                data-testid="search-bar"
                name="search"
                placeholder="movie name.."
                type="text"
                value=""
              />
              <svg
                class="search-icon"
                fill="none"
                stroke="white"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="mt-4 w-full"
        >
          <div>
            <div
              class="z-10 flex h-screen items-center justify-center bg-black"
              title="loading"
            >
              <svg
                class="h-12 w-12 animate-spin"
                data-testid="loading-icon"
                fill="none"
                stroke="white"
                stroke-width="1"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0l3.181 3.183a8.25 8.25 0 0013.803-3.7M4.031 9.865a8.25 8.25 0 0113.803-3.7l3.181 3.182m0-4.991v4.99"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
